pyhf/modifiers/__init__.py

```diff
@@ -11,7 +11,7 @@
 Check if given object contains the right structure for constrained and unconstrained modifiers
 '''
 def validate_modifier_structure(modifier, constrained):
-    required_methods = ['__init__', 'add_sample', 'apply']
+    required_methods = ['__init__']
```

Comment:
this can probably be entirely removed (checking for existence of `__init__` doesn't make much sense